### ISSUE TITLE
F-Droid Android App can no longer sync. 

### ISSUE THREAD

**Issue by develroo** [dani-garcia/bitwarden_rs#1023]
So this started a few weeks back and I thought it was just a temporary glitch with the app but no for some reason all sync seem to fail even though I can see data being transferred over the network.

The set up is this. bitwarden_rs installed via docker and connected to over a VPN. I have now set up a self-signed SSL thinking that the app had perhaps increased it's security like that. The logs show nothing save normal connections. 
```

[2020-06-05 20:19:54][bitwarden_rs::api::identity][INFO] User user@my.email.net logged in successfully. IP: 10.4.0.99
[2020-06-05 20:19:54][response][INFO] POST /identity/connect/token (login) => 200 OK
[2020-06-05 20:19:56][request][INFO] GET /api/sync
[2020-06-05 20:19:57][response][INFO] GET /api/sync?<data..> (sync) => 200 OK
[2020-06-05 20:20:07][request][INFO] GET /api/sync
[2020-06-05 20:20:07][response][INFO] GET /api/sync?<data..> (sync) => 200 OK
[2020-06-05 20:20:54][request][INFO] GET /api/sync
[2020-06-05 20:20:55][response][INFO] GET /api/sync?<data..> (sync) => 200 OK
[2020-06-05 20:21:20][request][INFO] GET /api/sync
[2020-06-05 20:21:20][response][INFO] GET /api/sync?<data..> (sync) => 200 OK

```
For reference the Android app is 2.4.0 and the docker version of bitwarden_rs is 1.15.0-db535118  

Any idea how I go about debugging this issue? Like I say I can see it starts transferring data but stops after about 3 seconds say there is a sync error. 

The Firefox addon and Web portal still work fine though ?

**Comment by BlackDex** [comment]
First, which android app are you using. I can not find a bitwarden app at all. Let alone the official one on F-Droid. The official app can be download from https://play.google.com/store/apps/details?id=com.x8bit.bitwarden .

Second, self-signed certificates are know to cause some issues on mobile devices like Android or iOS.
You need to import the CA into your browser or mobile device to allow the apps and browser to connect to it without any issue.

There is an example on how to do this here: https://community.bitwarden.com/t/self-signed-certificate-on-local-network-works-with-chrome-iphone-and-android/2676

If you need to build the CA again, you could try to use https://github.com/FiloSottile/mkcert to generate this.

Also keep in mind that Apple is going to distrust certificates which are valid for more then one year.
I don't know what other vendors or browsers are going to do with this.

**Comment by develroo** [comment]
I have done all that. The CA work fine. 

There is a version of Bitwarden in Fdroid. It can be found [here](https://mobileapp.bitwarden.com/fdroid/) however I just read the changelog [here](https://github.com/bitwarden/mobile/releases) and realise there was an update for the app to 2.4.2. So I followed their instructions and uninstalled and re-installed the app. It was odd because it was updated only 4 days ago.

Anyway.. seems with the 2.4.2 update it is syncing again now. So it was the app's fault not the server install.

Feel free to close this issue. Thanks.
